nixos/modules/services/misc/sourcehut/default.nix

```diff
@@ -0,0 +1,168 @@
+      mkKeyValue = key: v: let
+        isPath = v: builtins.typeOf v == "path";
+
+        value =
+          if null == v        then ""
+          else if true == v   then "yes"
+          else if false == v  then "no"
+          else if isInt v     then toString v
+          else if isString v  then toString v
+          else if isPath v    then toString v
+          else abort "sourcehut.mkKeyValue: unexpected type (v = ${v})";
+      in "${toString key}=${value}";
+    in generators.toINI { inherit mkKeyValue; } cfg.settings;
+
+    environment.systemPackages = [ python ];
+
+    # PostgreSQL server
+    services.postgresql.enable = true;
+    # Mail server
+    services.postfix.enable = mkOverride 999 true;
+    # Cron daemon
+    services.cron.enable = mkOverride 999 true;
+    # Redis server
+    services.redis = {
+      enable = true;
+      port = mkDefault 6379;
+      # TODO: localhost
+      bind = mkDefault "127.0.0.1";
+      # TODO: More like 2?
+      databases = mkDefault 8;
+    };
+
+    services.sourcehut.settings = {
+      # The name of your network of sr.ht-based sites
+      "sr.ht".site-name = mkDefault "sourcehut";
+      # The top-level info page for your site
+      "sr.ht".site-info = mkDefault "https://sourcehut.org";
+      # {{ site-name }}, {{ site-blurb }}
+      "sr.ht".site-blurb = mkDefault "the hacker's forge";
+      # If this != production, we add a banner to each page
+      "sr.ht".environment = mkDefault "development";
+      # Contact information for the site owners
+      "sr.ht".owner-name = mkDefault "Drew DeVault";
+      "sr.ht".owner-email = mkDefault "sir@cmpwn.com";
+      # The source code for your fork of sr.ht
+      "sr.ht".source-url = mkDefault "https://git.sr.ht/~sircmpwn/srht";
+      # A secret key to encrypt session cookies with
+      "sr.ht".secret-key = mkDefault null;
+
+      # Outgoing SMTP settings
+      mail.smtp-host = mkDefault null;
+      mail.smtp-port = mkDefault null;
+      mail.smtp-user = mkDefault null;
+      mail.smtp-password = mkDefault null;
+      mail.smtp-from = mkDefault null;
+      # Application exceptions are emailed to this address
+      mail.error-to = mkDefault null;
+      mail.error-from = mkDefault null;
+      # Your PGP key information (DO NOT mix up pub and priv here)
+      # You must remove the password from your secret key, if present.
+      # You can do this with gpg --edit-key [key-id], then use the passwd
+      # command and do not enter a new password.
+      mail.pgp-privkey = mkDefault null;
+      mail.pgp-pubkey = mkDefault null;
+      mail.pgp-key-id = mkDefault null;
+
+      # base64-encoded Ed25519 key for signing webhook payloads. This should be
+      # consistent for all *.sr.ht sites, as we'll use this key to verify signatures
+      # from other sites in your network.
+      #
+      # Use the srht-webhook-keygen command to generate a key.
+      webhooks.private-key = mkDefault null;
```

Comment:
Serious security problem here.
Any idea? I thought of just appending this part from the content of a secure file, creating a third, non-store config for the service to use.

Reply:
IIRC services e.g. ssh generate keys on service start if they're not present...

Reply:
Appending won't fix the problem, it'll still end up in the store. If you don't want it to be in the store, the only option is to have as an external file and you're better off setting it up yourself since I can't access the settings unless I parse the file's contents.

I'm pretty sure the generated key isn't suppose to be used for ssh connections.

Reply:
Why wont appending fix the problem? I dont think you understood what I meant. The file we're appending from is *not* in the store, and neiter is the file were appending to. Only the 'base' file (the one created here, without private keys) would be in the store.

Reply:
As far as I know, the only configuration file for sourcehut lies in `/etc/sr.ht/config.ini` since the relative `config.ini` is out of the question unless one starts overriding the derivation.

```
a = 1;
${builtins.readFile path_to_external_file}
```
Would have the file contents of path_to_external_file in the file in the store.

Reply:
This really is a sourcehut bug. Secrets dont go in config files.